### PR TITLE
Makefile target to build and push image for release

### LINKED ISSUE #135: Compile aws-cloud-controller-manager during prow CI
**What would you like to be added**:
Our current prow CI is very limited in that it only runs unit tests and a few linters, we should also compile aws-cloud-controller-manager. 

**Why is this needed**:
To catch any compile errors that may not be caught in unit tests.

/kind feature

### PR DESCRIPTION
**Pull request by nckturner**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:

We want to build and push containers using prow post-submits.  This is a first pass at a script that takes care of tag format validation and passing GCR credentials to docker.  Simplified version of the vsphere cloud provider release.sh script and makefile.  

**Which issue(s) this PR fixes**:

Fixes #135

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,17 +12,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+.EXPORT_ALL_VARIABLES:
```
Comment (randomvariable) [inline Makefile:15]: TIL

**Comment by randomvariable** [comment]
For an example of the cloudbuild config, we have:
the cloud build configuration:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/cloudbuild.yaml
Makefile: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/Makefile
Dockerfile: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/Dockerfile

Image promoter configuration:
Staging: https://github.com/kubernetes/k8s.io/tree/master/k8s.gcr.io/manifests/k8s-staging-cluster-api-aws
Promoted images: https://github.com/kubernetes/k8s.io/tree/master/k8s.gcr.io/images/k8s-staging-cluster-api-aws
Actual staging repo: https://console.cloud.google.com/gcr/images/k8s-staging-cluster-api-aws/GLOBAL

Release process:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/book/src/development/releasing.md

**Comment by randomvariable** [comment]
Looking good. I think we can iterate once we have cloudbuild enabled in test-infra?

**Comment by nckturner** [comment]
> Looking good. I think we can iterate once we have cloudbuild enabled in test-infra?

Yup.

**Review by M00nF1sh — APPROVED** [review]
/lgtm

**Pull request merged**